src/puppetlabs/jdbc_util/pool.clj

```diff
@@ -65,12 +65,12 @@
           (loop []
             (if-let [result
                      (try
-                       (with-open [conn (.getConnection datasource)]
-                         (try (init-fn conn)
-                              (catch Exception e
-                                (swap! init-error (constantly e))
-                                (log/errorf e "%s - An error was encountered during initialization." (.getPoolName datasource))))
-                         datasource)
+                       (.close (.getConnection datasource))
```

Comment:
Might be worth adding a clarifying comment to explain what it is doing.  On first read through it took a second to understand the point.